articles/building-a-nodejs-application-using-docker/index.md

```diff
@@ -0,0 +1,241 @@
+# Building A Node.js Application Using Docker
+
+[Docker](https://www.docker.com/) allows us to run our applications as *containers*. A container is a standalone executable package which is lightweight and has everything needed to run an application be it libraries, tools, runtime, setting or code.
+
+In this tutorial, we will build a Node.js application, create an image and also build a container using the image. Enjoy!!!
+
+
+# Dockerize the Node.js Application
+
+To start this tutorial, we need to make sure Docker and [Node.js](https://nodejs.org/en/download/package-manager/) is installed on our system. If Docker is not, you can refer to the links below to download:
+- [Ubuntu](https://docs.docker.com/engine/installation/linux/ubuntu/)
+- [Windows](https://docs.docker.com/docker-for-windows/install/)
+- [MacOS](https://docs.docker.com/docker-for-mac/install/)
+
+In order to dockerize a Node.js Application we need to follow these steps:
+
+- Creating A Node.js Application
+- Create A DockerFile
+- Building your Docker Image
+- Expose
+
+# Step 1 - Creating a Node.js Application
+
+First, we will start by creating a directory for our project then install some dependencies for our simple Hello World Website.
+```
+ mkdir node-web-app
+ cd node-web-app
+```
+
+### Install NPM and Express Framework
+
+Install npm and install the express framework, which is one of Node.js frameworks. Initialize npm in our directory.
```

Comment:
Remove second install for readability. Which is **a** Node.js framework would also flow better as well as **Then,** initialize